DigitalLearningSolutions.Data.Tests/Helpers/PostLearningAssessmentTestHelper.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.Helpers
+{
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.PostLearningAssessment;
+    using Microsoft.Data.SqlClient;
+
+    public class PostLearningAssessmentTestHelper
+    {
+        private SqlConnection connection;
+
+        public PostLearningAssessmentTestHelper(SqlConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public IEnumerable<OldPostLearningAssessmentScores> ScoresFromOldStoredProcedure(int progressId, int sectionId)
```

Comment:
We're only expecting this to return one lot of scores right? So it would probably make more sense to call `FirstOrDefault` here, instead of `Where` 🙂

Reply:
The stored procedure takes a `ProgressID` and returns a row for each section of the customisation while the method in our service also takes the specific `SectionID` and returns a single row. The `Where` is being used to retrieve an individual section from the stored procedure in order to compare the values to the ones from the service.

Reply:
Using `FirstOrDefault` would return the first row from the stored procedure, but it may not be the section that we want to compare.

Reply:
You can provide a comparison function to FirstOrDefault I believe, like you do with Where. So you should be able to use that instead of having to use FirstOrDefault later 🙂 https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.firstordefault?view=net-5.0